pkg/server/httpAPI.go

```diff
@@ -466,6 +476,32 @@ func (h *HTTPHandler) serveConfig(rw http.ResponseWriter, req *http.Request) {
 	h.Logger.Info("Served config for ETCD instance.")
 }
 
+// GetClusterState returns the Cluster state either `new` or `existing`.
+func (h *HTTPHandler) GetClusterState(ctx context.Context, clusterSize int, clientSet client.Client, podName string, podNS string) string {
+	if clusterSize == 1 {
+		return miscellaneous.ClusterStateNew
+	}
+
+	// clusterSize > 1
+	state := miscellaneous.GetInitialClusterState(ctx, *h.Logger, clientSet, podName, podNS)
+	if len(state) == 0 {
+		// Not a Scalup scenario.
+		// Either a multi-node bootstarp or a restoration of single member in multi-node.
+		m := member.NewMemberControl(h.EtcdConnectionConfig)
+
+		// check for a learner presence in a cluster.
+		// if a learner is present then return ClusterStateExisting else ClusterStateNew.
+		present, _ := m.IsLearnerPresent(ctx)
```

Comment:
Can you please also read the error from the function call and act on it accordingly?

Reply:
yes, will do

Reply:
done. PTAL